code/modules/mob/living/carbon/human/species_types/oozelings.dm

```diff
@@ -0,0 +1,156 @@
+		var/obj/item/clothing/CS = H.wear_suit
+		var/obj/item/clothing/CH = H.head
+		if(CS.clothing_flags & CH.clothing_flags & STOPSPRESSUREDAMAGE)
+			atmos_sealed = TRUE
+	if(H.w_uniform && H.head)
+		var/obj/item/clothing/CU = H.w_uniform
+		var/obj/item/clothing/CH = H.head
+		if (CU.envirosealed && (CH.clothing_flags & STOPSPRESSUREDAMAGE))
+			atmos_sealed = TRUE
+	if(!atmos_sealed)
+		var/datum/gas_mixture/environment = H.loc.return_air()
+		if(environment)
+			if(environment.total_moles())
+				if(environment.get_moles(/datum/gas/water_vapor) >= 1)
+					H.blood_volume -= 15
+					if(prob(50))
+						to_chat(H, "<span class='danger'>Your ooze melts away rapidly in the water vapor!</span>")
+				if(environment.get_moles(/datum/gas/plasma) >= 1)
+					if(H.blood_volume <= 672)
+						H.blood_volume += 15
+	if(H.blood_volume < BLOOD_VOLUME_OKAY)
+		if(prob(5))
+			to_chat(H, "<span class='danger'>You feel drained!</span>")
+	if(H.blood_volume < BLOOD_VOLUME_BAD)
+		Cannibalize_Body(H)
+	if(regenerate_limbs)
+		regenerate_limbs.UpdateButtonIcon()
+
+/datum/species/oozeling/proc/Cannibalize_Body(mob/living/carbon/human/H)
+	var/list/limbs_to_consume = list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG) - H.get_missing_limbs()
+	var/obj/item/bodypart/consumed_limb
+	if(!limbs_to_consume.len)
+		H.losebreath++
+		return
+	if(H.get_num_legs(FALSE)) //Legs go before arms
+		limbs_to_consume -= list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM)
+	consumed_limb = H.get_bodypart(pick(limbs_to_consume))
+	consumed_limb.drop_limb()
+	to_chat(H, "<span class='userdanger'>Your [consumed_limb] is drawn back into your body, unable to maintain its shape!</span>")
+	qdel(consumed_limb)
+	H.blood_volume += 80
+	H.nutrition += 20
+
+/datum/action/innate/regenerate_limbs
+	name = "Regenerate Limbs"
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "slimeheal"
+	icon_icon = 'icons/mob/actions/actions_slime.dmi'
+	background_icon_state = "bg_alien"
+
+/datum/action/innate/regenerate_limbs/IsAvailable()
+	if(..())
+		var/mob/living/carbon/human/H = owner
+		var/list/limbs_to_heal = H.get_missing_limbs()
+		if(limbs_to_heal.len && H.blood_volume >= BLOOD_VOLUME_OKAY+40)
+			return TRUE
+		return FALSE
+
+/datum/action/innate/regenerate_limbs/Activate()
+	var/mob/living/carbon/human/H = owner
+	var/list/limbs_to_heal = H.get_missing_limbs()
+	if(!LAZYLEN(limbs_to_heal))
+		to_chat(H, "<span class='notice'>You feel intact enough as it is.</span>")
+		return
+	to_chat(H, "<span class='notice'>You focus intently on your missing [limbs_to_heal.len >= 2 ? "limbs" : "limb"]...</span>")
+	if(H.blood_volume >= 40*limbs_to_heal.len+BLOOD_VOLUME_OKAY)
+		if(do_after(H, 60, target = H))
+			H.regenerate_limbs()
+			H.blood_volume -= 40*limbs_to_heal.len
+			to_chat(H, "<span class='notice'>...and after a moment you finish reforming!</span>")
+		return
+	else if(H.blood_volume >= 40)//We can partially heal some limbs
```

Comment:
You shouldn't really need an else if, as the if statement above it returns if true anyway :)

Reply:
oof, thanks!
fixed on next push